common/csv.go

```diff
@@ -0,0 +1,69 @@
+package common
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+)
+
+func (j *JSONHTTPClient) PutCSV(ctx context.Context, url string, reqBody []byte, headers ...Header) ([]byte, error) {
+	fullURL, err := j.getURL(url)
+	if err != nil {
+		return nil, err
+	}
+
+	_, body, err := j.httpPutCSV(ctx, fullURL, headers, reqBody) // nolint:bodyclose
+	if err != nil {
+		return nil, err
+	}
+
+	return body, nil
+}
```

Comment:
Sorry, it was my suggestion to keep this as JSONHTTPClient for now since the HTTP response from the bulk write API is still in JSON format, and it's helpful to have the JSON parsing functionality still. Using the HTTP client directly means that the JSON response parsing functionality would have to be re-implemented. I do agree this is a bit confusing still, and we should come up with a better strategy here, whether that's renaming this client or creating a different client. Since you're in flight today and this PR has already been outstanding for a while and is blocking the server changes, let's merge it in, and do the refactoring in a follow up.